doc/content/guides/getting-started/configuration.md

```diff
@@ -4,29 +4,158 @@ description: ""
+Next, we'll have a look at the configuration options for your private deployment. These options are set in the `environment` section of the `stack` service in `docker-compose.yml`.
+
+First we'll make sure that The Things Stack uses the correct databases.
+
+```
+TTN_LW_IS_DATABASE_URI=postgres://root@cockroach:26257/ttn_lorawan?sslmode=disable
+TTN_LW_REDIS_ADDRESS=redis:6379
+```
+
+Then we'll configure TLS with Let's Encrypt. Since we're deploying The Things Stack on `your.thethingsstack.io`, we configure it to only request certificates for that host, and to also use it as the default host.
+
+```
+TTN_LW_TLS_ACME_ENABLE=true
+TTN_LW_TLS_ACME_DIR=/var/lib/acme
+TTN_LW_TLS_ACME_EMAIL=your@email.com
+TTN_LW_TLS_ACME_HOSTS=your.thethingsstack.io
+TTN_LW_TLS_ACME_DEFAULT_HOST=your.thethingsstack.io
+```
+
+Next, we'll configure the HTTP server with keys for encrypting and verifying cookies, and with passwords for endpoints that you may want to keep for internal use.
+
+```
+TTN_LW_HTTP_COOKIE_HASH_KEY=generate 64 bytes (openssl rand -hex 64)
+TTN_LW_HTTP_COOKIE_BLOCK_KEY=generate 32 bytes (openssl rand -hex 32)
+TTN_LW_HTTP_METRICS_PASSWORD=choose a password
+TTN_LW_HTTP_PPROF_PASSWORD=choose a password
+```
 
-ca: /path/to/your/cert.pem
+The Things Stack sends emails to users, so we need to configure how those are sent. 
+
+```
+TTN_LW_IS_EMAIL_SENDER_NAME=The Things Stack
+TTN_LW_IS_EMAIL_SENDER_ADDRESS=noreply@your.thethingsstack.io
+TTN_LW_IS_EMAIL_NETWORK_CONSOLE_URL=https://your.thethingsstack.io/console
+TTN_LW_IS_EMAIL_NETWORK_IDENTITY_SERVER_URL=https://your.thethingsstack.io/oauth
+```
+
+You can either use Sendgrid or an SMTP server. If you don't set up an email provider, The Things Stack will print emails to the server log.
+
+```
+TTN_LW_IS_EMAIL_PROVIDER=sendgrid
+TTN_LW_IS_EMAIL_SENDGRID_API_KEY=enter Sendgrid API key
+```
+
+or
+
+```
+TTN_LW_IS_EMAIL_PROVIDER=smtp
+TTN_LW_IS_EMAIL_SMTP_ADDRESS=enter mail server address
+TTN_LW_IS_EMAIL_SMTP_USERNAME=enter username
+TTN_LW_IS_EMAIL_SMTP_PASSWORD=enter password
+```
+
+Finally, we need to configure the Web UI to use `your.thethingsstack.io`.
+
+```
+TTN_LW_IS_OAUTH_UI_CANONICAL_URL=https://your.thethingsstack.io/oauth
+TTN_LW_IS_OAUTH_UI_IS_BASE_URL=https://your.thethingsstack.io/api/v3
+
+TTN_LW_CONSOLE_OAUTH_AUTHORIZE_URL=https://your.thethingsstack.io/oauth/authorize
+TTN_LW_CONSOLE_OAUTH_TOKEN_URL=https://your.thethingsstack.io/oauth/token
+
+TTN_LW_CONSOLE_UI_CANONICAL_URL=https://your.thethingsstack.io/console
+TTN_LW_CONSOLE_UI_AS_BASE_URL=https://your.thethingsstack.io/api/v3
+TTN_LW_CONSOLE_UI_GS_BASE_URL=https://your.thethingsstack.io/api/v3
+TTN_LW_CONSOLE_UI_IS_BASE_URL=https://your.thethingsstack.io/api/v3
+TTN_LW_CONSOLE_UI_JS_BASE_URL=https://your.thethingsstack.io/api/v3
+TTN_LW_CONSOLE_UI_NS_BASE_URL=https://your.thethingsstack.io/api/v3
+
+TTN_LW_CONSOLE_OAUTH_CLIENT_ID=console
+TTN_LW_CONSOLE_OAUTH_CLIENT_SECRET=generate a good one (openssl rand -hex 32)
```

Comment:
Indicate that this is hex and the length?

I can implicitly deduce that from the `openssl` command, but maybe something like `enter generated 32-bit hex secret ($ openssl rand -hex 32)`

Reply:
Not hex, and doesn't matter. I'll reword.